qiskit/tools/qi/pauli.py

```diff
@@ -41,262 +42,487 @@ class Pauli:
                 np.array(x.to_label(), 'c') == b'I'))
-
-        elif case == 1:
+        elif case == 'tensor':
             # the Pauli set is in tensor order II IX IY IZ XI ...
-            for k_index in range(4 ** number_of_qubits):
-                v = np.zeros(number_of_qubits)
-                w = np.zeros(number_of_qubits)
+            for k in range(4 ** number_of_qubits):
+                v = np.zeros(number_of_qubits, dtype=np.bool)
+                w = np.zeros(number_of_qubits, dtype=np.bool)
                 # looping over all the qubits
-                for j_index in range(number_of_qubits):
-                    # making the Pauli for each kindex i fill it in from the
+                for j in range(number_of_qubits):
+                    # making the Pauli for each j fill it in from the
                     # end first
-                    element = int((k_index) / (4 ** (j_index))) % 4
-                    if element == 0:
-                        v[j_index] = 0
-                        w[j_index] = 0
-                    elif element == 1:
-                        v[j_index] = 0
-                        w[j_index] = 1
+                    element = (k // (4 ** j)) % 4
+                    if element == 1:
+                        w[j] = True
                     elif element == 2:
-                        v[j_index] = 1
-                        w[j_index] = 1
+                        v[j] = True
+                        w[j] = True
                     elif element == 3:
-                        v[j_index] = 1
-                        w[j_index] = 0
+                        v[j] = True
                 temp_set.append(Pauli(v, w))
             return temp_set
+        else:
+            raise QISKitError("Only support 'weight' or 'tensor' cases")
 
-    print('please set the number of qubits to less than 5')
-    return -1
+    raise QISKitError("Only support number of qubits is less than 5")
 
 
 def pauli_singles(j_index, number_qubits):
     """Return the single qubit pauli in number_qubits."""
     # looping over all the qubits
     tempset = []
-    v = np.zeros(number_qubits)
-    w = np.zeros(number_qubits)
-    v[j_index] = 0
-    w[j_index] = 1
+    v = np.zeros(number_qubits, dtype=np.bool)
+    w = np.zeros(number_qubits, dtype=np.bool)
+    w[j_index] = True
     tempset.append(Pauli(v, w))
-    v = np.zeros(number_qubits)
-    w = np.zeros(number_qubits)
-    v[j_index] = 1
-    w[j_index] = 1
+    v = np.zeros(number_qubits, dtype=np.bool)
+    w = np.zeros(number_qubits, dtype=np.bool)
+    v[j_index] = True
+    w[j_index] = True
     tempset.append(Pauli(v, w))
-    v = np.zeros(number_qubits)
-    w = np.zeros(number_qubits)
-    v[j_index] = 1
-    w[j_index] = 0
+    v = np.zeros(number_qubits, dtype=np.bool)
+    w = np.zeros(number_qubits, dtype=np.bool)
+    v[j_index] = True
     tempset.append(Pauli(v, w))
     return tempset
+
+
+if __name__ == '__main__':
+    p = Pauli.from_label('ZIII')
```

Comment:
what is this below

Reply:
just for my test, will remove it later, and I will add test_pauli.py for testing